### PR TITLE
Lifecycle aware

### PR DESCRIPTION
**Pull request by tharmaman**
Implemented LifecycleAware and attached it to the easyDAO.

Lifecycles consist of four stages: PENDING, ACTIVE, REJECTED and DELETED.

Viewing REJECTED and DELETED objects will require special permissions and under the LifecycleAwareDAO, a remove_ gets overridden with a put_ that sets the objects lifecycleState to DELETED similar to DeletedAware,

### CODE REVIEW

**Review by jlhughes — APPROVED** [review]

File: src/foam/dao/EasyDAO.js [hunk 1]
```diff
@@ -194,6 +194,13 @@ foam.CLASS({
             .build();
         }
 
+        if ( getLifecycleAware() ) {
```
Comment (lchanmann) [inline src/foam/dao/EasyDAO.js]: Can we have lifecycleAware before deletedAware? and log a deprecated warning in the deletedAware block.

Comment (nanoscott) [reply src/foam/dao/EasyDAO.js]: Can both be enabled at the same time? If not, we should throw an error if both are set to true ..

**Review by nanoscott — APPROVED** [review]

File: src/foam/dao/EasyDAO.js (hunk 1 above)
Comment (tharmaman) [reply src/foam/dao/EasyDAO.js]: Done to both.

**Comment by lchanmann** [comment]
LGTM, thanks @tharmaman.

**Pull request merged**